### PR TITLE
Update frontend

### PR DESCRIPTION
**Pull request by jordisala1991**
Main things done here:
* Use NPM instead of Yarn
* Use a fixed Node version (16) managed through .nvmrc
* Update devDependencies and replace deprecated ones
* Update docs and Github Actions

There is no need to keep using Yarn, NPM is better and more updated (Yarn is already at v3)

## Subject

I am targeting this branch, because this is BC and does not produce any change for end users, so it is pedantic

### CODE REVIEW

**Comment by VincentLanglet** [comment]
> There is no need to keep using Yarn, NPM is better and more updated (Yarn is already at v3)

I always used yarn so far and when I looked at article on internet about yarn vs npm, yarn seems to be better.
Do you have some informations about this ?

**Comment by jordisala1991** [comment]
https://www.mixmax.com/engineering/to-yarn-and-back-again-npm

Tl;dr: Yarn was better a long ago, now there is almost no difference. NPM comes installed on all node versions but yarn needs to be enabled/ installed. Also we are using yarn 1 while there is already yarn 3, which complicated things with the plug'n'play feature (forcing you to disable it to keep using node modules).

IMO there is no real disadvantage of using NPM.

**Comment by jordisala1991** [comment]
Now, if you want a faster alternative to npm/yarn (they are really similar in performance) you go for pnpm. Which is way faster than both (but again a package that is not pre installed)

File: assets/scss/styles.scss [hunk 1]
```diff
@@ -136,7 +136,7 @@ div {
   }
 
   &.mosaic-inner-box-hover {
-    background-color: rgba(255, 255, 255, 0.8);
+    background-color: rgb(255 255 255 / 80%);
```
Comment (VincentLanglet) [inline assets/scss/styles.scss:139]: I would have prefer `,` as separator more but not a blocker

File: assets/scss/tree.scss [hunk 2]
```diff
@@ -62,7 +62,7 @@
     border: 1px solid #3c8dbc;
 
     &::after {
-      border-color: rgba(255, 255, 255, 0);
+      border-color: rgb(255 255 255 / 0%);
```
Comment (VincentLanglet) [inline assets/scss/tree.scss:65]: same

File: assets/scss/tree.scss [hunk 3]
```diff
@@ -85,7 +85,7 @@
     }
 
     &::before {
-      border-color: rgba(255, 255, 255, 0);
+      border-color: rgb(255 255 255 / 0%);
```
Comment (VincentLanglet) [inline assets/scss/tree.scss:88]: same

**Review by VincentLanglet — APPROVED** [review]

File: assets/scss/styles.scss (hunk 1 above)
Comment (jordisala1991) [reply assets/scss/styles.scss:139]: It is a stylelint thing: https://stylelint.io/user-guide/rules/list/color-function-notation/

apparently the `,` is an older thing, the new notation is this one

**Comment by jordisala1991** [comment]
To merge with https://github.com/sonata-project/dev-kit/pull/1927 (to avoid a dev-kit PR reverting some of this)
